Makefile

```diff
@@ -23,24 +27,13 @@ backup:
 clean:
 # remove test containers and restore test/config directory
 	-@ [[ -d testconfig.bak ]] && { sudo rm -rf test/config ; mv testconfig.bak test/config ; } || :
-	-@ for container in $$(docker ps -a --filter name='^/mail$$|^ldap_for_mail$$|^mail_override_hostname$$|^mail_non_subdomain_hostname$$|^open-dkim$$|^hadolint$$|^eclint$$|^shellcheck$$|mail_changedetector.*' | sed 1d | cut -f 1-1 -d ' '); do docker rm -f $$container; done
-	-@ sudo rm -rf test/onedir test/alias test/quota test/relay test/config/dovecot-lmtp/userdb test/config/key* test/config/opendkim/keys/domain.tld/ test/config/opendkim/keys/example.com/ test/config/opendkim/keys/localdomain2.com/ test/config/postfix-aliases.cf test/config/postfix-receive-access.cf test/config/postfix-receive-access.cfe test/config/dovecot-quotas.cf test/config/postfix-send-access.cf test/config/postfix-send-access.cfe test/config/relay-hosts/chksum test/config/relay-hosts/postfix-aliases.cf test/config/dhparams.pem test/config/dovecot-lmtp/dh.pem test/config/relay-hosts/dovecot-quotas.cf test/config/user-patches.sh test/alias/config/postfix-virtual.cf test/quota/config/dovecot-quotas.cf test/quota/config/postfix-accounts.cf test/relay/config/postfix-relaymap.cf test/relay/config/postfix-sasl-password.cf test/duplicate_configs/
+	-@ for CONTAINER in $$(docker ps -a --filter name='^dms-test-.*|^hadolint$$|^eclint$$|^shellcheck$$' | sed 1d | cut -f 1-1 -d ' '); do docker rm -f $${CONTAINER}; done
+	-@ while read -r LINE; do [[ $${LINE} =~ test/.+ ]] && rm -rf $${LINE}; done < .gitignore
```

Comment:
I like this, smart :sunglasses: 

Is this going to be ok with `test/config/key*`? The `.gitignore` also has some mixed usage of a trailing `/` for directories from the looks of it. I assume none of that is an issue though?

There may be some irrelevant lines we're cleaning up on now, I think the `letsencrypt` one (_in the `.gitignore` may no longer be used in tests_), but we can do some house keeping on that at another time :+1: 

`sudo` might be needed still? If these are files being created from a container mount, then it may be writing to the disk as the `root` user / group on the host. If it's just stuff tests are generating without that issue though, not needing `sudo` would be great :grinning:

Reply:
> I like this, smart sunglasses

:D

> Is this going to be ok with `test/config/key*`? The `.gitignore` also has some mixed usage of a trailing `/` for directories from the looks of it. I assume none of that is an issue though?

I double-checked it, we should be fine. This is using something that `shellcheck` would actually complain about: Bash "globbing" the star at the end :) This is what we want and need. I also made sure only directories have trailing slashes in `.gitignore`.
 
> There may be some irrelevant lines we're cleaning up on now, I think the `letsencrypt` one (_in the `.gitignore` may no longer be used in tests_), but we can do some house keeping on that at another time +1

Removed the LE line, kept others there. Another housekeeping PR can solve this, exactly :)

> `sudo` might be needed still? If these are files being created from a container mount, then it may be writing to the disk as the `root` user / group on the host. If it's just stuff tests are generating without that issue though, not needing `sudo` would be great grinning

Indeed, I added `sudo` to the `rm` call.